app/src/main/java/com/amazonaws/kinesisvideo/webrtc/URLCapturer.java

```diff
@@ -0,0 +1,100 @@
+package com.amazonaws.kinesisvideo.webrtc;
+
+import android.content.Context;
+import android.net.Uri;
+
+import org.videolan.libvlc.LibVLC;
+import org.videolan.libvlc.Media;
+import org.videolan.libvlc.MediaPlayer;
+import org.videolan.libvlc.interfaces.IVLCVout;
+import org.webrtc.CapturerObserver;
+import org.webrtc.SurfaceTextureHelper;
+import org.webrtc.VideoCapturer;
+import org.webrtc.VideoFrame;
+import org.webrtc.VideoSink;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This class implements a VideoCapturer for media streams. Similar to how camera capturers work, this
+ * class also leverages the SurfaceTextureHelper to do the heavy lifting of grabbing and encoding video
+ * frame using an intermediate OpenGL texture. It also uses libVLC(User must add) to playback streams on the texture.
+ */
+public class URLCapturer implements VideoCapturer, VideoSink {
+    private String url;
+    private String[] options;
+    private String aspectRatio;
+    private SurfaceTextureHelper surfaceTextureHelper;
+    private Context context;
+    private CapturerObserver capturerObserver;
+
+    /**
+     * Public constructor that accepts the stream url.
+     * @param url Media stream url.
+     */
+    public URLCapturer(String url, String[] options, String aspectRatio){
+        this.url = url;
+        this.options = options;
+        this.aspectRatio = aspectRatio;
+    }
+
+    @Override
+    public void initialize(SurfaceTextureHelper surfaceTextureHelper, Context context, CapturerObserver capturerObserver) {
+        this.surfaceTextureHelper = surfaceTextureHelper;
+        this.context = context;
+        this.capturerObserver = capturerObserver;
+
+        surfaceTextureHelper.startListening(this);
+    }
+
+    @Override
+    public void onFrame(VideoFrame videoFrame) {
+        capturerObserver.onFrameCaptured(videoFrame);
+    }
+
+    @Override
+    public void startCapture(int width, int height, int framerate) {
+        capturerObserver.onCapturerStarted(true);
+        surfaceTextureHelper.setTextureSize(width, height);
+
+        // Use libVLC to play the stream onto the texture.
+        LibVLC libVlc = new LibVLC(context, Arrays.asList(options));
+
+        MediaPlayer mediaPlayer = new MediaPlayer(libVlc);
+        IVLCVout vOut = mediaPlayer.getVLCVout();
+        vOut.setWindowSize(width, height);
+        vOut.setVideoSurface(surfaceTextureHelper.getSurfaceTexture());
```

Comment:
Thanks for calling this out. The usecase we had didn't need audio and so it was not implemented. But I agree that audio needs to be handled. Let me check how this may be added. Please let me know if you have any suggestions on how to add audio support as well.

Reply:
Applied this change. Kindly review.